conda/core/solve.py

```diff
@@ -1363,9 +1421,9 @@ def _pin_python(self, state, update_modifier=NULL):
             return f"python {major_minor_version}.*"
 
     def _pinned_specs(self):
-        pinned_specs = get_pinned_specs(context.target_prefix)
+        pinned_specs = get_pinned_specs(self.prefix)
```

Comment:
Could you elaborate why this change?

Reply:
Some tests do pass the prefix explicitly to the class (which defaults to `context.target_prefix` if `prefix` is not passed explicitly).